### DISCUSSION TITLE
WindFarm_Wake

### DISCUSSION THREAD

**Discussion by oguzcrs11** [OpenFAST/openfast#2351] [Q&A] [answered]
Dear all;

My name is Oğuzhan, and I am currently a Master's student in Turkey. I have been working on analyzing fatigue for downstream turbines, particularly focusing on accounting for wake effects. However, due to my limited experience with OpenFAST, I have been progressing step by step and have encountered some challenges.

From my research, I understand that simulation data and outputs for the NREL 5MW turbine have been used for fatigue analysis with MLife. However, I have not been able to find exercises or datasets that involve multiple turbines. At this point, I am looking for a wind farm dataset that includes wake effects for downstream turbines, which I can use in QBlade or Ashes for further analysis or "it might be more accurate to say wind data".

As I am still learning how to connect the information I have gathered, any guidance or resources you could provide would be invaluable. Additionally, I plan to use QBlade or Ashes for visualization since I am not fully familiar with Paraview.

Please let me know if any part of my request is unclear or if there are specific details you need. I greatly appreciate your time and assistance.

Thank you for your consideration.

Best regards,

**Comment by jjonkman** [comment]
Dear @oguzcrs11,

I'm not sure I fully understand what you are asking, but it sounds like you want turbulent inflow for a given wind turbine that takes into consideration not only ambient inflow turbulence, but also the turbulence from upstream wind turbine wakes.

The IEC design standard 61400-1 outlines a simple "effective turbulence" model that could be applied to this, which works well on average, but misses details associated with the wind farm layout and operational conditions.

For the later, tools such as FAST.Farm can be used to model wind turbine aeroelastics in OpenFAST combined with farm-level wake and array effects from multiple turbines in the farm.  But if you were applying FAST.Farm, then you would be using OpenFAST rather than QBlade or Ashes.

FYI: We recently wrote a paper comparing FAST.Farm against the IEC effective turbulence model, which may be informative to you: https://wes.copernicus.org/articles/8/1475/2023/.

Best regards,

**Reply by oguzcrs11** [reply]
Dear @jjonkman,

Thank you very much for the information provided. It was exactly what I needed to learn at this stage "not only ambient inflow turbulence, but also the turbulence from upstream wind turbine wakes.". I appreciate your guidance on this matter, as well as informing me about the related paper.

I am eager to review the paper and delve deeper into the subject. Your assistance has been invaluable.

Thank you once again for your support.

Best regards,

**Reply by oguzcrs11** [reply]
Dear @jjonkman;
When I want to run FAST.Farm.fstf file, I encounter an error (in image), Can you help me with the cause of the error?
![Ekran görüntüsü 2024-08-06 214526](https://github.com/user-attachments/assets/d7735a91-e0b6-4876-bacd-435f64e41159)

I checked the all input files's path , is it true ? 
I've already install from this link the files: https://github.com/OpenFAST/r-test/tree/main 

At the same time, also FFTest_WT1.fst gives error:
![Ekran görüntüsü 2024-08-06 215159](https://github.com/user-attachments/assets/c2809980-e211-4129-a465-7c425393f060)
but as outputs VTP files are created.

I can't change anything anyway, but the notepad content is like this
![Ekran görüntüsü 2024-08-06 215456](https://github.com/user-attachments/assets/fb85f224-5a91-44c3-a0de-2aef156047c2)
![Ekran görüntüsü 2024-08-06 215644](https://github.com/user-attachments/assets/67b7c516-3f30-4c93-ad0e-0e5804873bdb)
Thank you very much in advance, I look forward to your answer.

Best Regards.

**Reply by jjonkman** [reply]
Dear @oguzcrs11,

The FAST.Farm error is triggered by a formatting issue with your FAST.Farm input (_*.fstf_) file.  You can always enable the `Echo` file to debug issues with input file formatting, but in your case the error is obvious--you have specified the wind farm has one turbine (NumTurbines = 1), but then list data for two turbines, resulting an an extra line in the input file.

The OpenFAST error is triggered because it can't find the turbulent wind input file.  But this OpenFAST model is set up to work with FAST.Farm, not with standalone OpenFAST.

Best regards,

**Reply by oguzcrs11** [reply]
Dear @jjonkman 
Firstly, thanks for advice, it worked but now:
![image](https://github.com/user-attachments/assets/43a7b439-6cc4-4af7-bcca-6ca21f6ad39a)

As I understand it, the problem is in 90m_08mps.inp but what is the meaning of column width of 10 instead of 20 characters. 
![image](https://github.com/user-attachments/assets/4a8ccd93-e7ae-4091-9570-3efaa37dce82)

Best Regards.

**Reply by jjonkman** [reply]
Dear @oguzcrs11,

The message about the column width is a warning, which is independent from the error listed on the next line that is actually causing FAST.Farm to abort.

The error appears to be triggered because a TurbSim binary output file should be specified via input `FileName_BTS` in InflowWind, but instead, `FileName_BTS` is pointing to a TurbSim input file.

Best regards,

**Reply by oguzcrs11** [reply]
Dear @jjonkman, 

With your advice, I handled many problem in this section. Thanks for all.

But now, When I run it, it gives a new error:
![image](https://github.com/user-attachments/assets/a711f222-47c6-4e59-b21c-6cb9d9f8f58e)

In this section, Iguess wind turbine2  is out of of bounds of the high-resolution domain in the global Y position. 
I want to ask a question about this.

If I want to think that the turbines are positioned on a flat surface, Should I set all values ​​to 0 [blue marked]?

![image](https://github.com/user-attachments/assets/234c063a-d90e-48a4-b365-78c3ad913ce1)
Thanks for your time

Best Regards.

**Reply by jjonkman** [reply]
Dear @oguzcrs11,

I would guess the issue is that the 2nd wind turbine is not within the domain in the X direction, given that the `WT_X` differs between the turbines while `X0_High` does not.  The high-resolution domain of each wind turbine must fully encompass that turbine, including all possible displacements of the turbine.

Best regards,

**Reply by oguzcrs11** [reply]
Dear @jjonkman 

There's something on my mind. In my solution I didn't include the ServoDyn, when I examined the result of fastfarm variables like OoPDefl1	IPDefl1 or Turbine power result I obtained values ​​that I thought would be unrealistic. In Servodyn I see a variable entry called Bladed-style DLL.

I may not have much of an idea about what the healthy operating conditions of a turbine would be.

Best Regards.

**Reply by jjonkman** [reply]
Dear @oguzcrs11,

I'm not sure I really understand your question.  Active control through ServoDyn is important in most operational conditions.  If you are using the NREL 5-MW baseline wind turbine, I would recommend using the provided Bladed-style DISCON controller DLL unless you are using your own controller.

Best regards,

**Reply by oguzcrs11** [reply]
Dear @jjonkman,
Even though I can't explain it, you understand it very well :)
I used the Discon.dll from example files.
by the way try to setup an onshore NREL-5MW

I'm not sure if this is a problem but I got this error:
....
Running ElastoDyn.
Nodal outputs section of ElastoDyn input file not found or improperly formatted.
Running AeroDyn.
Running InflowWind.
Running ServoDyn.
Running ServoDyn Interface for Bladed Controllers (using Intel Visual Fortran for Windows).
Using legacy Bladed DLL interface.

Farm_Initialize:Farm_ValidateInput:OutFmt produces a column width of 10 instead of 20 characters.
Farm_Initialize:For efficiency, NumPlanes has been reduced to the number of time steps (19).

 Time: 96 of 150 seconds.  Estimated final completion at 21:06:37.
The BEM solution is being turned off due to low TSR.  (TSR = 1.9976). This warning will not be
 Time: 128 of 150 seconds.  Estimated final completion at 21:06:38.

T2:FARM_UpdateStates:FWrap_Increment:FAST_Solution:FAST_AdvanceStates:AD_UpdateStates:BEMT_UpdateS
tates:UpdatePhi(node 9, blade 2):BEMT_UnCoupledSolve:There is no valid value of phi for these
operating conditions: Vx = 4.9351, Vy = -34.229, rlocal = 28.131, theta = 0.13615, geometric phi
= 2.9984. This warning will not be repeated though the condition may persist. (See GeomPhi output
channel.)

Best Regards.

**Reply by jjonkman** [reply]
Dear @oguzcrs11,

The FAST.Farm message you share does not include any fatal errors that will stop the simulation, but warnings about "low TSR" and "no valid value of phi" may be problematic if you are not expecting them for the simulation set up you are using.  Are the responses of the two turbines expected for the conditions you are simulating?

Best regards,

**Reply by oguzcrs11** [reply]
Dear @jjonkman,

for ELASTODYN v1.03.* INPUT FILE, BldNodes are specified numerically in elastodayn, but how are these blades nodes input? 
![image](https://github.com/user-attachments/assets/95c74808-85d6-4a32-9b92-c79ca0529d63)
I found a simple file about this subject  : 
![image](https://github.com/user-attachments/assets/65ac5a6b-b9f2-4777-9973-deb8b210fd58)
I thought that just stating the number (as in the first photo) was not enough, I noticed that this part can also be adjusted in different wind turbine analysis programs.

Can the file path be specified in the part where the number is?
Thank you again
Best Regards.

**Reply by jjonkman** [reply]
Dear @oguzcrs11,

In ElastoDyn, the blade structural analysis nodes are placed automatically without additional user input.  That is, the blade (between the root and tip) is discretized into `BldNodes` equal-length elements and the blade structural analysis nodes are located in the centers of those elements such that the first node is located at 0.5*(`TipRad`-`HubRad`)/`BldNodes` + `HubRad` along the pitch axis from the hub centerline, the second node is located at 1.5*(`TipRad`-`HubRad`)/`BldNodes` + `HubRad`, etc. for each node from 1 to `BldNodes`.  These blade structural analysis node locations are written to the ElastoDyn summary (_*.ED.sum_) file.

In AeroDyn, the blade aerodynamic analysis nodes are user-located and these nodes can be placed independently from the structural analysis nodes.

Best regards,

**Reply by oguzcrs11** [reply]
Dear @jjonkman,

If I'm not mistaken, BEMT is extension of Actuator disk and many actuator method is in the literature, Is it possible to add another method instead of BEMT in OpenFast e.g. actuator line.
Please excuse me if my question is unnecessary. Just I want to compare the results of different methods.

Best Regards.

**Reply by jjonkman** [reply]
Dear @oguzcrs11,

I agree that the "momentum" part of BEMT relies on actuator disk assumptions, although the "element" part of BEMT relies on an actuator line.

OpenFAST itself can already bypass actuator disk assumptions by switching from BEMT to the free-vortex wake method (in the OLAF submodel of AeroDyn).

Moreover, OpenFAST has also been coupled to high-fidelity CFD through an actuator-line formulation (e.g., in OpenFOAM (SOWFA) and AMR-Wind) and through a geometry-resolved formulation (e.g,. in NaluWind).

Best regards,

**Reply by oguzcrs11** [reply]
Dear @jjonkman,

Is it possible to use cfd in Fast or should I use SOWFA for it. I would like to add a chapter in my thesis  a title  like that "comparsion of the blade element momentum with cfd results" for single turbine.

Best Regards.

**Reply by jjonkman** [reply]
Dear @oguzcrs11,

High-fidelity CFD is not directly available within OpenFAST, but is available through an actuator-line formulation (e.g., in the coupling between OpenFAST and OpenFOAM (SOWFA) or AMR-Wind) and through a geometry-resolved formulation (e.g,. in the coupling between OpenFAST and NaluWind).

Best regards,

**Comment by oguzcrs11** [comment]
Dear @jjonkman,

All your reply was very helpful for me thanks a lot.
I don't know how necessary it is to compare elastodyn and beamdyn results, but I saw that there was an article and frequent questions about this in the past. based on your past answers I applied condition of numcrctn:1 and I started to study similar results between two results with this condition. In NREL form your reply was  "I would compare the deflections using the “xb” axis in ElastoDyn against the “xr” axis in BeamDyn." and I printed this values on chart but I didn't understand why for beamdyn result showed 0

![image](https://github.com/user-attachments/assets/2fbc3c6a-4436-4d70-9d90-8e7fe6e7ed12) 
![image](https://github.com/user-attachments/assets/9218c8c5-71f9-40fb-8fdd-e10a31729d40)
![image](https://github.com/user-attachments/assets/44d60ea0-ec2f-4e96-9717-94a33bedadf3)

e.g. RootFxb1-B1RootFxr 
![{071BCDA5-18D9-46AA-A4B9-EE53E948FD54}](https://github.com/user-attachments/assets/73525807-5f2c-4d5d-a2e4-2e2858e43512)
all beamdyn result =0 for root forces and moments

Best Regards.

**Reply by jjonkman** [reply]
Dear @oguzcrs11,

Can you clarify what you are selecting for the blade root outputs from BeamDyn in BeamDyn's `OutList`?  Does the OpenFAST output file say that the blade root outputs from BeamDyn you've selected are "INVALID" (in which case OpenFAST will output zeros)?

Best regards,

**Comment by oguzcrs11** [comment]
I selected the variables under the heading of reaction loads according to the outlistparameters in the excel file;
![image](https://github.com/user-attachments/assets/555e2fcc-0004-4312-83bb-928d9ae9ce92)
![image](https://github.com/user-attachments/assets/85181977-8fec-487e-9732-664da466ad71)

everything else is the same with rtest 
![image](https://github.com/user-attachments/assets/0b232d92-2ab7-45df-b771-0c3b5e754172)

Is there a chance it's related to fast dt?
![image](https://github.com/user-attachments/assets/5779c17d-7ebd-4a20-a01a-941b80291445)

Best Regards.

**Reply by jjonkman** [reply]
In the OpenFAST output (_*.out_) file, are the units of these blade-root loads showing up correctly, or are they listed as "INVALID"?  Are you sure you are grabbing the correct column of data in the output file?

Best regards,

**Reply by oguzcrs11** [reply]
![{86693F4B-BC4F-4C4F-9447-056F2DF68897}](https://github.com/user-attachments/assets/e4bce292-de53-4997-b129-701424b3f4d6)
I never thought to look at the invalid units in the out file but why, Did I write the wrong variables but these are same with at excell 
Is there a step you can suggest?

Best Regards.

**Reply by jjonkman** [reply, marked as answer]
My guess is that you are grabbing the ElastoDyn outputs (`RootFxb1`, etc.), which are invalid, rather than the BeamDyn outputs (`B1RootFxr`, etc.).

Best regards,
